lib/spack/docs/getting_started.rst

```diff
@@ -926,75 +926,33 @@ Once ``curl`` has been installed, you can similarly install the others.
 Environment Modules
 """""""""""""""""""
 
-In order to use Spack's generated environment modules, you must have
-installed one of *Environment Modules* or *Lmod*.  On many Linux
-distributions, this can be installed from the vendor's repository.  For
-example: ``yum install environment-modules`` (Fedora/RHEL/CentOS). If
-your Linux distribution does not have Environment Modules, Spack can
-build it for you!
+In order to use Spack's generated module files, you must have
+installed one of *Environment Modules* or *Lmod*. The preferred method
+to get the latest version of either of these tools is installing
+it with Spack, for instance:
 
-What follows are three steps describing how to install and use environment-modules with spack.
-
-#. Install ``environment-modules``.
-
-   * ``spack bootstrap`` will build ``environment-modules`` for you (and may build
-     other packages that are useful to the operation of Spack)
-
-   * Install ``environment-modules`` using ``spack install`` with
-     ``spack install environment-modules~X`` (The ``~X`` variant builds without Xorg
-     dependencies, but ``environment-modules`` works fine too.)
-
-#. Add ``modulecmd`` to ``PATH`` and create a ``module`` command. 
-
-   * If you are using ``bash`` or ``ksh``, Spack can currently do this for you as well.
-     After installing ``environment-modules`` following the step
-     above, source Spack's shell integration script. This will automatically
-     detect the lack of ``modulecmd`` and ``module``, and use the installed
-     ``environment-modules`` from ``spack bootstrap`` or ``spack install``.
-     
-     .. code-block:: console
-
-        # For bash/zsh users
-        $ export SPACK_ROOT=/path/to/spack
-        $ . $SPACK_ROOT/share/spack/setup-env.sh
-
-
-   * If you prefer to do it manually,  you can activate with the following 
-     script (or apply the updates to your ``.bashrc`` file manually):
+.. code-block:: console
 
-         .. code-block:: sh
+   $ spack install lmod
```

Comment:
shouldn't we just advertise `spack bootstrap` here? It will install a proper module system *and* all other missing dependencies, no?

Reply:
I am hesitant for only one reason: `spack bootstrap` installs `environment-modules` at the moment. This means that it won't work out of the box for anybody who wants to use Lmod module hierarchies. Maybe you're right and we should advertize only `spack bootstrap`, and extend it later to prefer `lmod`.

Reply:
One off-topic question: if `spack install lmod` is enough to make `lmod` known to spack, shouldn't `spack install gcc@5.4.0` be enough to add it to `compilers.yaml` automatically? :) https://github.com/spack/spack/issues/5101#issuecomment-324013176

Reply:
@ax3l I should have more time to devote to Spack's development going to the end of the year. Simplify the management of compilers is on the radar. :smirk: